lib/src/commands/updates.dart

```diff
@@ -0,0 +1,45 @@
+import 'package:args/command_runner.dart';
+import 'package:yaml/yaml.dart';
+
+import '../api.dart';
+import '../pubspec.dart';
+
+class UpdatesCommand extends Command {
+  @override
+  final String name = 'updates';
+
+  @override
+  final List<String> aliases = ['u'];
+
+  @override
+  final String description =
+      'Checks pub.dev for packages in your pubspec that have updated versions';
+
+  @override
+  final String invocation = 'pubx updates';
+
+  Future<void> run() async {
+    final pub = findPubspec();
+
+    // Load and parse pubspec.yaml
+    final pubContents = await pub.readAsStringSync();
+    final pubYaml = loadYaml(
+      pubContents,
+      sourceUrl: pub.uri,
+    ) as YamlMap;
+
+    final YamlMap dependencies = pubYaml['dependencies'];
+
+    dependencies.keys.forEach((packageName) async {
+      if (!dependencies[packageName].toString().startsWith('{sdk:')) {
+        try {
+          final pkgInfo = await fetchPackageInfo(packageName);
+          print(
+              '${pkgInfo.name} [${dependencies[packageName]}]  latest: ${pkgInfo.version}');
```

Comment:
I notice in the PR description the strings are aligned in columns, but don't appear to be here.

Reply:
It's fine if this is intentional. We can always evolve it. Just curious whether this was intentional.

Reply:
yes sorry about that, I intended for them aligned but didn't get time to do that when I first wrote this PR, I'll have a go at getting that done now.

Reply:
I've reworked this a bit so the output should be correctly aligned, taking into account package name lengths and hosted vs non-hosted packages, eg.

```
provider               	 [^3.0.0+1] 	 latest: 4.0.2
uuid                   	 [^2.0.1] 	 latest: 2.0.4
device_info            	 [^0.4.0] 	 latest: 0.4.1+4
package_info           	 [^0.4.0] 	 latest: 0.4.0+13
http                   	 [path pkg] 	 latest: 0.12.0+4
sentry                 	 [git pkg] 	 latest: 3.0.1
flutter_markdown       	 [git pkg] 	 latest: 0.3.3
url_launcher           	 [^5.1.4] 	 latest: 5.4.1
google_sign_in         	 [^4.0.1+3] 	 latest: 4.1.1
retry                  	 [^3.0.0] 	 latest: 3.0.0+1
cupertino_icons        	 [^0.1.2] 	 latest: 0.1.3
flutter_facebook_login 	 [^3.0.0] 	 latest: 3.0.0

```

Reply:
👍